recipe/meta.yaml

```diff
@@ -8,36 +8,44 @@ package:
 source:
   url: https://lhapdf.hepforge.org/downloads/?f=LHAPDF-{{ version }}.tar.gz
   sha256: 2443a4b32cc3b0597c8248bd6e25703ace9c91a7a253c5f60b1b5428ef9c869e
+  patches:
+    # Apply patch to wrappers/python/build.py.in to build bindings with Cython
+    # https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/wrappers/python/build.py.in
+    - osx_patch.diff  # [osx]
 
 build:
   skip: true  # [win]
   skip: true  # [python_impl == "pypy"]
 
-  number: 4
+  number: 5
   detect_binary_files_with_prefix: true
 
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython <3                              # [build_platform != target_platform]
+    - cython >=3.0.0  # [build_platform != target_platform]
     - gnuconfig  # [unix]
     - {{ compiler('cxx') }}
     - {{ stdlib("c") }}
     - make
-    - automake  # [build_platform != target_platform]
-    - libtool   # [build_platform != target_platform]
+    - automake
+    - autoconf
+    - libtool
```

Comment:
Why is it necessary to remove the `# [build_platform != target_platform]`? These were not needed in Linux before I think.

Reply:
@chrisburr removed these, but the pattern wasn't necessary. `# [build_platform != target_platform]` is only useful when you're cross compiling and need the `.so`s.

c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#details-about-cross-compiled-python-packages

Reply:
But as far as I am aware these are not used or needed in the linux (or mac for that matter) builds right?

Reply:
I needed it for a native macOS build due to: https://github.com/conda-forge/lhapdf-feedstock/blob/d89b118c527edee53db55b2fbb372be52411c9dd/recipe/build.sh#L11

Potentially the requirement should be the same in the `build.sh`. Though I'd also suggest always cythonising the sources as there isn't much of a downside and it could be useful in cases like new python versions.

Reply:
This PR now changes to using the GitLab source for build uniformity, which requires these now, and so gives the same experience that @chrisburr had. It is also important to point out that in the future there could and probably will be native Apple silicon runners on Azure pipelines, and so this means that future builds will work on those native architecture runners.

Reply:
You mean you needed it when running the conda-recipe locally?

I'd really prefer to keep the build/installation as close to the "official installation" as possible.

Reply:
> This PR now changes to using the GitLab source for build uniformity, which requires these now,

Ah, gotcha. I missed that when writing the comment above. That resolves the question.

I need to think about moving from hepforge to gitlab because I'm not convinced, but that clarifies most of my questions / points.

Reply:
> You mean you needed it when running the conda-recipe locally?

No, @chrisburr means building on an aarch64 macOS machine. If you get the source (GitLab or hepforce) you are required to generate a configure script using `autoreconf` (or `autoconf`) as the GitLab source doesn't provide a `configure` script and the hepforge provided `configure` script doesn't know how to support `aarch64` macOS. In both situations a native build on aarch64 macOS requires doing the "normal" autotools projects procedure of generating `configure`.

Reply:
> No, @chrisburr means building on an aarch64 macOS machine.

Which must've beeen local (or at least, not the CI)!

In any case, I was missing the gitlab part (which either wasn't here yesterday or I missed it) to see it was needed now also in linux.

Reply:
> Which must've been local (or at least, not the CI)!

Ah yes, correct. Sorry I didn't make that clear.

> I was missing the gitlab part (which either wasn't here yesterday or I missed it) to see it was needed now also in linux.

You're correct that was a today change. I apologize about the rebasing as that can make some of the tracking of changes harder to follow across time, but easier to look at if you're looking at each commit in a review. I'll update the PR body and then also leave comments in a "review" about how and why the changes were made, but to make things less noisy I'll not do any further rebasing until there's a consensus on what direction to move things towards. :+1:

Reply:
No worries, I should've waited until it was finished to look through the changes!